src/view/xdg-shell.cpp

```diff
@@ -338,7 +338,8 @@ std::string wayfire_xdg_view::get_title()
 
 void wayfire_xdg_view::close()
 {
-    wlr_xdg_surface_send_close(xdg_surface);
+    wlr_xdg_popup_destroy(xdg_surface);
```

Comment:
This function should just do a `wlr_xdg_toplevel_send_close` (views aren't popups in any case). For popups, we should add an override `wayfire_xdg_popup::close` which calls the `wlr_xdg_popup_destroy`, but that isn't necessary until #110  is resolved.